cmd/ebrelayer/txs/relayToEthereum.go

```diff
@@ -22,47 +23,57 @@ const (
 	GasLimit = uint64(3000000)
 )
 
+var (
+	gasUsedLastTx uint64
+)
+
 // RelayProphecyClaimToEthereum relays the provided ProphecyClaim to CosmosBridge contract on the Ethereum network
 func RelayProphecyClaimToEthereum(provider string, contractAddress common.Address, event types.Event,
-	claim ProphecyClaim, key *ecdsa.PrivateKey) error {
+	claim ProphecyClaim, key *ecdsa.PrivateKey, cethAmount *big.Int) (uint64, error) {
+
 	// Initialize client service, validator's tx auth, and target contract address
 	client, auth, target, err := initRelayConfig(provider, contractAddress, event, key)
 	if err != nil {
-		return err
+		return 0, err
+	}
+
+	// If ceth amount is lower than estimated gas
+	if cethAmount.Cmp(big.NewInt(int64(gasUsedLastTx))) < 0 {
+		return 0, errors.New("not enough ceth to cover the gas costs")
```

Comment:
Instead of looking at the gas used last time, I think we should instead look at the current gas price because ethereum gas prices vary so much depending on network demand.

I think we should scope out how much gas these 3 transactions take in total, use that as a constant, then get the current gas price of the network, and use that to calculate how much ceth needs to be burned.

Reply:
Just use the gas used last time as reference and we can't calculate the total cost since other two transaction happened in other relayer nodes. So the tx fee ceth we charged is not precise but very close.

Reply:
We do have an accurate reading of how much gas is used for each type of transaction, so let's use that here. I don't like the idea of being imprecise as it could lead to deficits especially in the early days of peggy when there is a few transactions going across the bridge to ethereum. This would lead to incorrect amounts of gas being paid especially because ethereum gas prices swing so rapidly.

When we mint tokens on the ethereum side, it costs us 282031 gas in total for 3/4 validators to send this tx.
When we unlock or send eth on the etheruem side, it costs us 248692 gas. Based on this, we can check the tx type, and charge the gas accordingly.